File: app/src/main/java/com/example/win/easy/dagger/module/DownloadModule.java
package com.example.win.easy.dagger.module;

import android.os.Environment;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class DownloadModule {

    @Provides @Named("downloadRootDir")static String provideDownloadRootDir(){
        return Environment.getExternalStorageDirectory().getAbsolutePath()+"/Swift Switch";
    }

    @Provides @Named("suffixOfTempFile")static String provideSuffixOfTempFile(){
        return "~TEMP";
    }
}

File: app/src/main/java/com/example/win/easy/dagger/module/DownloadModule.kt
package com.example.win.easy.dagger.module

import android.os.Environment
import com.example.win.easy.download.DownloadService
import com.example.win.easy.download.DownloadServiceAdapter
import com.example.win.easy.download.FileService
import com.example.win.easy.repository.SongRepository
import dagger.Module
import dagger.Provides
import kotlinx.coroutines.ExecutorCoroutineDispatcher
import javax.inject.Named
import javax.inject.Singleton

@Module
object DownloadModule {

    @JvmStatic
    @Provides
    @Named("downloadRootDir")
    fun provideDownloadRootDir() = Environment.getExternalStorageDirectory().absolutePath + "/Swift Switch"

    @JvmStatic
    @Provides
    @Named("suffixOfTempFile")
    fun provideSuffixOfTempFile() = "~TEMP"

    @JvmStatic
    @Provides
    @Singleton
    fun provideFileService() = FileService()

    @JvmStatic
    @Provides
    @Singleton
    fun provideDownloadService(fileService: FileService, songRepository: SongRepository) = DownloadService(fileService, songRepository)

    @JvmStatic
    @Provides
    @Singleton
    fun provideDownloadServiceAdapter(downloadDispatcher: ExecutorCoroutineDispatcher, downloadService: DownloadService) = DownloadServiceAdapter(downloadDispatcher,downloadService)
}